internal/locate/region_cache.go

```diff
@@ -1723,6 +1730,59 @@ func (c *RegionCache) GetTiFlashStores() []*Store {
 	return stores
 }
 
+// GetTiFlashMPPStores returns all stores with lable <engine, tiflash_mpp>.
+func (c *RegionCache) GetTiFlashMPPStores(bo *retry.Backoffer) (res []*Store, err error) {
+	c.tiflashMPPStoreMu.RLock()
+	needReload := c.tiflashMPPStoreMu.needReload
+	stores := c.tiflashMPPStoreMu.stores
+	c.tiflashMPPStoreMu.RUnlock()
+
+	if needReload {
+		return c.reloadTiFlashMPPStores(bo)
+	}
+	return stores, nil
+}
+
+func (c *RegionCache) reloadTiFlashMPPStores(bo *retry.Backoffer) (res []*Store, _ error) {
+	stores, err := c.pdClient.GetAllStores(bo.GetCtx())
+	if err != nil {
+		return nil, err
+	}
+	mppLabels := []*metapb.StoreLabel{
+		{
+			Key:   "engine",
+			Value: "tiflash_mpp",
```

Comment:
Better to use constants, I find some [similar usage](https://github.com/tikv/client-go/blob/79f010d806110cb4bccf9218ceae5e5db54951fe/tikvrpc/endpoint.go#L62-L67) here.